### ISSUE TITLE
Alternate assembly folders specified using the command line option -f is not honoured by runner

### ISSUE THREAD

**Issue by ramprasathdgl** [jediwhale/fitsharp#156]
Hi,

I have been trying to provide  the alternate folder path for the assemblies using the commad line opttion -f as shown below

>  ./runner.exe -c suite.config.xml 9999 -f alternateassemblypath1,alternateassemblypath2

The runner searches for the dll specified in the ApplicationUnderTest but on failure does not search the alternate folder paths specified using the command line option -f

### Below is my findings on debugging the issue FYI

The runner.exe registers for the [ System.AppDomain.CurrentDomain.AssemblyResolve](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=netframework-4.7.2) event if the command line option -f .
The  [ System.AppDomain.CurrentDomain.AssemblyResolve](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=netframework-4.7.2)  event is fired only when the assembly is failed to load by the API System.Reflection.Assembly.Load .

The Loading of assembly in runner.exe is handle by the [fitSharp.Machine.Engine.LoadAssembly](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.load?view=netframework-4.7.2)  method , In this method the assembly is loaded using the .Net API [Assembly.LoadFrom ](https://github.com/jediwhale/fitsharp/blob/master/source/fitSharp/Machine/Engine/CurrentDomain.cs#L13) instead of [System.Reflection.Assembly.Load API](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.load?view=netframework-4.7.2) as a result the [ System.AppDomain.CurrentDomain.AssemblyResolve](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.assemblyresolve?view=netframework-4.7.2)  event is not fired and the Assembly resolve handler is not called and the path to the dll is not resolved

**Comment by jediwhale** [comment]
I'll take a look.  That feature was contributed by someone else and I've never used it!

**Comment by ramprasathdgl** [comment]
Cool, Thanks for looking in to it

**Comment by jediwhale** [comment]
This is what I found:

The -f option does not provide alternate folders for finding assemblies that you load directly - the assumption is that you know where these assemblies are and will provide the correct path.  The -f option is used to provide alternate folders for finding assemblies referenced by the assemblies that you load.  I tested this and it works.

e.g.  I load an assembly from c:\my\bin by specifying the path c:\my\bin\mycode.dll.  This assembly references myutils.dll which is in c:\other\bin, so I use -f c:\other\bin.
